### ISSUE TITLE
Marketplace license violation during Code-OSS build?

### ISSUE THREAD

**Issue by randomir** [microsoft/vscode#141340]
According to #81173, use of Microsoft Marketplace from Code-OSS forks is in violation of the Marketplace terms.

But since [some extensions](https://github.com/microsoft/vscode/blob/bb6916effbb5a54b45cd54cdc7473a526d7b0672/product.json#L37-L98) are [downloaded](https://github.com/microsoft/vscode/blob/64551de22ba6b595bd5008e436292ab84b0036c2/build/lib/builtInExtensions.js#L53) from [the marketplace](https://github.com/microsoft/vscode/blob/64551de22ba6b595bd5008e436292ab84b0036c2/build/lib/extensions.js#L178) **during build**, isn't just building a (patched) fork in violation of the marketplace terms?

---

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: a wide range, including latest
- OS Version: any

Steps to Reproduce:

1. build vscode repo fork
2. built-in extensions are downloaded from the MS marketplace
3. terms violated?

**Comment by randomir** [comment]
Any updates on this one, @isidorn?

Ultimately, I would like to include **additional** bultin-in extensions in a 3rd party Code OSS build (pulling them from MS VS Marketplace **during build**), and I'm wondering if that's permissible. My interpretation (but I'm not a lawyer or license specialist) is that VS Marketplace forbids it -- but if that's correct, then bundling extensions during build in this repo (Code OSS) is **also** in violation of the license.

What am I missing? Thanks!

**Comment by randomir** [comment]
Sounds good, thank you @isidorn!

**Comment by xsrvmy** [comment]
You can clear `builtinExtensions` to not download them actually.

**Comment by xsrvmy** [comment]
Actually I think this could be considered a bug: the link in `build/lib/extensions.ts` should be configured in product.json instead, so that it can be changed to point to a different marketplace.

**Comment by randomir** [comment]
Exactly, I was surprised to see the marketplace [link is hard-coded](https://github.com/microsoft/vscode/blob/82dac48a4042d841173fe7028b82b3c3c176cfd9/build/lib/extensions.js#L178). But that can't be accidental, because [hard-coded `Client-Id` and `User-Id`](https://github.com/microsoft/vscode/blob/82dac48a4042d841173fe7028b82b3c3c176cfd9/build/lib/extensions.js#L169-L173) are sent in headers in API requests to the (hard-coded) marketplace endpoint.

So, either: (a) the distinction between build time and run time makes a difference for marketplace use/terms, or (b) this is a bug that causes violation of marketplace terms.

**Comment by xsrvmy** [comment]
The letter of the ToU states that it is running the extension downloaded from the marketplace in a fork of vscode that violates the marketplace terms. I have actually decided to file a bug against the marketplace because this causes an unclear licensing situation especially with GPL and CC licenses. The other approach, limiting the use of the API to VSCode itself, would not.

In regards to including extensions yourself, when you run vscode in development mode, there is a `--builtin` flag which allows you to change if you want to supply the extension via vscode marketplace, disable it, or bundle it locally. From there you can select a vsix file downloaded/compiled from github directly. Of course if you are going to build from source and want to add a lot of third-party extensions, you can always change the hard coded url.

**Comment by isidorn** [comment]
Built-in extensions being downloaded from the MS marketplace for Code OSS is not ok. It is a bug which we plan to fix. 
I will assign this item to the February milestone which means we will start looking into a solution. I will post updates here on the progress.
Thanks for raising this issue.

**Comment by isidorn** [comment]
We have started the investigation and work for this, but this will carry over to the next milestone, thus assigning to March.

**Comment by randomir** [comment]
Thank you for the update, @isidorn. Is there a PR that fixed this issue, that you can reference here (for posterity)? Thanks!

**Comment by isidorn** [comment]
Sure https://github.com/microsoft/vscode/pull/146863
